backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java

```diff
@@ -512,11 +513,14 @@ Processor executeObAction(Endpoint endpoint, UnaryOperator<Processor> updateFunc
     })
 
     @RolesAllowed(ConsoleIdentityProvider.RBAC_READ_INTEGRATIONS_ENDPOINTS)
-    public List<NotificationHistory> getEndpointHistory(@Context SecurityContext sec, @PathParam("id") UUID id, @QueryParam("includeDetail") Boolean includeDetail, @BeanParam Query query) {
-        // TODO We need globally limitations (Paging support and limits etc)
-        String orgId = getOrgId(sec);
-        boolean doDetail = includeDetail != null && includeDetail;
-        return notificationRepository.getNotificationHistory(orgId, id, doDetail, query);
+    public Page<NotificationHistory> getEndpointHistory(@Context SecurityContext sec, @PathParam("id") UUID id, @QueryParam("includeDetail") Boolean includeDetail, @BeanParam Query query) {
+        final String orgId = getOrgId(sec);
+        final boolean doDetail = includeDetail != null && includeDetail;
+
+        final List<NotificationHistory> notificationHistory = notificationRepository.getNotificationHistory(orgId, id, doDetail, query);
+        final long notificationHistoryCount = this.notificationRepository.countNotificationHistoryElements(id, orgId);
+
+        return new Page<>(notificationHistory, new HashMap<>(), new Meta(notificationHistoryCount));
```

Comment:
Could you use the `PageLinksBuilder` in here? 

https://github.com/RedHatInsights/notifications-backend/blob/0dfdcc6a585424a3f21b0f031245401c7d3c424f/backend/src/main/java/com/redhat/cloud/notifications/routers/NotificationResource.java#L93

Reply:
Sure, done!